### ISSUE TITLE
'FLAVR is not in the model registry'

### ISSUE THREAD

**Issue by ms421** [open-mmlab/mmediting#1003]
I am trying to train / test using FLAVR but i face this issue whether i try to train or test. I have only changed the flavr config file to my own dataset.

**Comment by bottledmind** [comment]
Same with TOFlow:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/mmcv/utils/registry.py", line 69, in build_from_cfg
    return obj_cls(**args)
  File "/content/mmediting/mmedit/models/video_interpolators/basic_interpolator.py", line 54, in __init__
    self.generator = build_backbone(generator)
  File "/content/mmediting/mmedit/models/builder.py", line 31, in build_backbone
    return build(cfg, BACKBONES)
  File "/content/mmediting/mmedit/models/builder.py", line 22, in build
    return build_from_cfg(cfg, registry, default_args)
  File "/usr/local/lib/python3.7/dist-packages/mmcv/utils/registry.py", line 62, in build_from_cfg
    f'{obj_type} is not in the {registry.name} registry')
KeyError: 'TOFlowVFI is not in the model registry'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "demo/video_interpolation_demo.py", line 89, in <module>
    main()
  File "demo/video_interpolation_demo.py", line 74, in main
    model = init_model(args.config, args.checkpoint, device=device)
  File "/content/mmediting/mmedit/apis/matting_inference.py", line 31, in init_model
    model = build_model(config.model, test_cfg=config.test_cfg)
  File "/content/mmediting/mmedit/models/builder.py", line 60, in build_model
    return build(cfg, MODELS, dict(train_cfg=train_cfg, test_cfg=test_cfg))
  File "/content/mmediting/mmedit/models/builder.py", line 22, in build
    return build_from_cfg(cfg, registry, default_args)
  File "/usr/local/lib/python3.7/dist-packages/mmcv/utils/registry.py", line 72, in build_from_cfg
    raise type(e)(f'{obj_cls.__name__}: {e}')
KeyError: "BasicInterpolator: 'TOFlowVFI is not in the model registry'"
```

**Comment by Season-Lee** [comment]
> Same with TOFlow:
> 
> ```
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.7/dist-packages/mmcv/utils/registry.py", line 69, in build_from_cfg
>     return obj_cls(**args)
>   File "/content/mmediting/mmedit/models/video_interpolators/basic_interpolator.py", line 54, in __init__
>     self.generator = build_backbone(generator)
>   File "/content/mmediting/mmedit/models/builder.py", line 31, in build_backbone
>     return build(cfg, BACKBONES)
>   File "/content/mmediting/mmedit/models/builder.py", line 22, in build
>     return build_from_cfg(cfg, registry, default_args)
>   File "/usr/local/lib/python3.7/dist-packages/mmcv/utils/registry.py", line 62, in build_from_cfg
>     f'{obj_type} is not in the {registry.name} registry')
> KeyError: 'TOFlowVFI is not in the model registry'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "demo/video_interpolation_demo.py", line 89, in <module>
>     main()
>   File "demo/video_interpolation_demo.py", line 74, in main
>     model = init_model(args.config, args.checkpoint, device=device)
>   File "/content/mmediting/mmedit/apis/matting_inference.py", line 31, in init_model
>     model = build_model(config.model, test_cfg=config.test_cfg)
>   File "/content/mmediting/mmedit/models/builder.py", line 60, in build_model
>     return build(cfg, MODELS, dict(train_cfg=train_cfg, test_cfg=test_cfg))
>   File "/content/mmediting/mmedit/models/builder.py", line 22, in build
>     return build_from_cfg(cfg, registry, default_args)
>   File "/usr/local/lib/python3.7/dist-packages/mmcv/utils/registry.py", line 72, in build_from_cfg
>     raise type(e)(f'{obj_cls.__name__}: {e}')
> KeyError: "BasicInterpolator: 'TOFlowVFI is not in the model registry'"
> ```

I meet the same problem. Have you solved the problem?

**Comment by bottledmind** [comment]
@Season-Lee 
Nah, I believe this repo is still in it's early stages of development. There are unpleasant bugs like these and it lacks documentation. Would be really nice to have at least a proper guide to inference all existing models via python, not scripts

**Comment by LeoXing1996** [comment]
Hey, @Season-Lee and @bottledmind, can you provide your config files, and then we can check what goes wrong?

**Comment by bottledmind** [comment]
@LeoXing1996 
I've got the previous outputs with default configs. Didn't modify or train a thing
`python demo/video_interpolation_demo.py ./configs/video_interpolators/flavr/flavr_in4out1_g8b4_vimeo90k_septuplet.py https://download.openmmlab.com/mmediting/video_interpolators/flavr/flavr_in4out1_g8b4_vimeo90k_septuplet_20220509-c2468995.pth /content/lowres_video.mp4 /content/out.mp4 --fps-multiplier 2 --device 0`
and
`python demo/video_interpolation_demo.py ./configs/video_interpolators/tof/tof_vfi_spynet_chair_nobn_1xb1_vimeo90k.py https://download.openmmlab.com/mmediting/video_interpolators/toflow/tof_vfi_spynet_chair_nobn_1xb1_vimeo90k_20220321-2fc9e258.pth /content/lowres_video.mp4 /content/out.mp4 --fps-multiplier 2 --device 0`

**Comment by LeoXing1996** [comment]
@Season-Lee We forget to import `FLAVR` in `mmedit.models.__init__.py` and `mmedit.models.video_interpolators.__init__.py`.

You can fix those two files as follow:
* `mmedit.models.video_interpolators.__init__.py`:
```diff
  from .basic_interpolator import BasicInterpolator
  from .cain import CAIN
+ from .flavr import FLAVR

- __all__ = ['BasicInterpolator', 'CAIN']
+ __all__ = ['BasicInterpolator', 'CAIN', 'FLAVR']
```

* `mmedit.models.__init__.py`
```diff
- from .video_interpolators import CAIN, BasicInterpolator
+ from .video_interpolators import CAIN, BasicInterpolator, FLAVR

 __all__ = [
     'AOTInpaintor', 'BaseModel', 'BasicRestorer', 'OneStageInpaintor', 'build',
     'build_backbone', 'build_component', 'build_loss', 'build_model',
     'BACKBONES', 'COMPONENTS', 'LOSSES', 'BaseMattor', 'DIM', 'MODELS',
     'GLInpaintor', 'PConvInpaintor', 'SRGAN', 'ESRGAN', 'GCA',
     'TwoStageInpaintor', 'IndexNet', 'DeepFillv1Inpaintor', 'Pix2Pix',
     'CycleGAN', 'SearchTransformer', 'LTE', 'FeedbackHourglass',
-   'BasicInterpolator', 'CAIN'
+   'BasicInterpolator', 'CAIN', 'FLAVR'
 ]
```

We fix this in PR #1186 as well. 

Please feel free to contact us if you find any bugs or meet any problems.
